src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -221,24 +249,169 @@ public void onTraversalStart() {
+                    throw new UserException("Error initializing GenomicsDBImporter for batch " + i +
+                            " processing ", e);
+                }
+
+
+                try {
+                    importer.importBatch();
+                    assert importer.isDone();
+                } catch (IOException e) {
+                    throw new UserException("GenomicsDB importBatch raised exception", e);
+                }
+
+                closeReaders(sampleToReaderMap);
+            }
         }
     }
 
     @Override
     public Object onTraversalSuccess() {
-        logger.info("Import completed!");
-        return importer.isDone();
+        if (batchSize==DEFAULT_ZERO_BATCH_SIZE) {
+            logger.info("Import completed!");
+        } else {
+            logger.info("Import of all batches to GenomicsDB completed!");
+        }
+        return true;
     }
 
-    @Override
-    public void closeTool() {
-        for (Map.Entry<String, FeatureReader<VariantContext>> reader : sampleToVCFMap.entrySet()) {
+    /**
+     * Method to create feature readers for input files or GCS URLs
+     * in the current batch
+     *
+     * @param sampleNames  List of all sample names
+     * @param sampleNameToFileName  Sample name to file name mapping
+     * @return  Feature readers to be imported in the current batch
+     */
+    private Map<String, FeatureReader<VariantContext>> getFeatureReaders(List<String> sampleNames, Map<String, String> sampleNameToFileName, int lowerSampleIndex, int upperSampleIndex) {
+        Map<String, FeatureReader<VariantContext>> sampleToReaderMap = new LinkedHashMap<>();
+
+        for(int i = lowerSampleIndex; i < upperSampleIndex; ++i) {
+            final String sampleName = sampleNames.get(i);
+            assert sampleNameToFileName.containsKey(sampleName);
+
+            final String variantURI = IOUtils.getPath(sampleNameToFileName.get(sampleName)).toAbsolutePath().toUri().toString();
+            final Function<SeekableByteChannel, SeekableByteChannel> cloudWrapper = (cloudPrefetchBuffer > 0 ? is -> SeekableByteChannelPrefetcher.addPrefetcher(cloudPrefetchBuffer, is) : Function.identity());
+            final Function<SeekableByteChannel, SeekableByteChannel> cloudIndexWrapper = (cloudIndexPrefetchBuffer > 0 ? is -> SeekableByteChannelPrefetcher.addPrefetcher(cloudIndexPrefetchBuffer, is) : Function.identity());
+            final AbstractFeatureReader<VariantContext, LineIterator> reader = AbstractFeatureReader.getFeatureReader(variantURI, null, new VCFCodec(), true, cloudWrapper, cloudIndexWrapper);
+
+            assert sampleName.equals(((VCFHeader) reader.getHeader()).getGenotypeSamples().get(0));
+            sampleToReaderMap.put(sampleName, reader);
+        }
+        return sampleToReaderMap;
+    }
+
+    /**
+     * Private method to create a GenomicsDB configuration data structure
+     * instead of sending a long list of parameters to the constructor call
+     *
```

Comment:
Add documentation for the method arguments. In particular, document what happens if you pass empty Strings in for the JSON files, and document whether the indices are inclusive or exclusive.